docs/memory.md

```diff
@@ -0,0 +1,104 @@
+- Snapshot - Display a list of current active memory objects (see Snapshot Classes).
+- Reset Accumulators - Reset values, to zero, under the column “Accumulator” in the Snapshot Classes list.
+- Filtering Classes - _TODO_
+- Snapshot Classes - Clicking on the Snapshot button, top right area, will display the list of current memory objects.  The memory objedts can be sorted by class name, size, allocated instances, etc.
+- Accumulator Counts Since Reset - Clicking on the Reset button, top right area, will reset the accumulated instances count.  Clicking on Snapshot after a reset will display the number of new instances allocated since last reset.  This is useful to finding memory leaks.
+- Instances of selected classes - Clicking on a class in the Snapshot Classes list will display the number of active instances.
+- Inspecting contents of an instance - _TODO_
+- Total Active Objects and Classes in the Heap - Total Classes allocated in the Heap and Total Objects (instances) in the Heap.
+
+## Memory Chart
+
+<img src="images/memory_basic_chart.png" width="800" />
+
+The chart's x-axis is a timeline of timestamps events (time series) from the VM as memory changes or the polled state of the memory
+every 500 ms.  This helps to give a live appearance on the state of the memory as the application is running.  The quantities plotted on the
+y-axis are (from top to bottom):
+- Capacity - Current capacity of the heap.
+- GC - GC has occurred.
+- Used - Objects (Dart objects) in the the heap.
+- External - memory not in the Dart heap, but is retained (memory read from a file) or decoded image in Flutter.
+
+<img src="images/memory_rss_chart.png" width="800" />
+
+To view RSS click on the RSS name in the legend.
+
+- RSS is the Resident Set Size and is used to show how much memory is allocated to that process and is in RAM. It does not include memory that is swapped out. It does include memory from shared libraries as long as the pages from those libraries are actually in memory. It does include all stack and heap memory.
+
+For more detailed information on the [Dart VM](https://mrale.ph/dartvm/) and memory.
+
+## Event Timeline
+
+<img src="images/memory_parts.png" width="800" />
+
+Chart displaying when a DevTool event (Snapshot and Reset button clicked) in relation to the timeline of the memory chart.  Hovering over the markers in the Event timeline will display the time when the Snapshot or Reset happened.  This helps to correlate to when in the timeline (x-axis) a memory leak might have occurred.
+
+Clicking on the Snapshot button will show the current state of the memory (heap) with regards to all active classes their instances.  When the Reset button is pressed the accumulator for all classes reset to zero, notice that the reset is tied with a faint blue when a Snapshot is again clicked displays the new accumulators values since the last Reset. 
```

Comment:
Clicking on the Snapshot button will show the current state of the heap with regard to all active classes and their instances. When the Reset button is pressed, the accumulator for all classes is reset to zero. The reset is temporally tied,  using a faint blue horizontal bar,  to the previous Snapshot. Clicking on the Reset button, again, will reset the accumulators since the last Reset and temporally tie the latest reset to the previous reset.